src/bluesearch/database/article.py

```diff
@@ -73,6 +76,121 @@ def paragraphs(self) -> Iterable[Tuple[str, str]]:
+        """Get the article title.
+
+        Returns
+        -------
+        str
+            The article title.
+        """
+        titles = self.content.find("//title-group/article-title")
+        return self.text_content(titles)
+
+    @property
+    def authors(self) -> Generator[str, None, None]:
+        """Get all author names.
+
+        Yields
+        ------
+        str
+            Every author, in the format "Given_Name(s) Surname".
+        """
+        authors = self.content.xpath('//contrib-group/contrib[@contrib-type="author"]')
+        for author in authors:
+            try:
+                given_names = self.text_content(author.find("name/given-names"))
+                surname = self.text_content(author.find("name/surname"))
+                author_str = given_names + " " + surname
+                yield author_str.strip()
+            except AttributeError:
+                # In rare cases, an author may not have a given name or a surname,
+                # e.g. it could be an organization. We decide to skip those.
+                continue
+
+    @property
+    def abstract(self) -> Generator[str, None, None]:
+        """Get a sequence of paragraphs in the article abstract.
+
+        Yields
+        ------
+        str
+            The paragraphs of the article abstract.
+        """
+        abstract_pars = self.content.findall("//abstract//p")
+        for paragraph in abstract_pars:
+            yield self.text_content(paragraph)
+
+    @property
+    def paragraphs(self) -> Generator[Tuple[str, str], None, None]:
+        """Get all paragraphs and titles of sections they are part of.
+
+        Paragraphs can be parts of text body, or figure or table captions.
+
+        Yields
+        ------
+        section : str
+            The section title.
+        text : str
+            The paragraph content.
+        """
+        # Paragraphs of text body
+        paragraphs = self.content.xpath("//body//p[not(parent::caption)]")
+        for paragraph in paragraphs:
+            text = self.text_content(paragraph)
+            section = paragraph.find("../title")
+            section_title = self.text_content(section)
+            yield section_title, text
+
+        # Figure captions
+        figs = self.content.findall("//body//fig") or []
+        for fig in figs:
+            fig_captions = fig.find("caption")
+            if fig_captions is None:
+                continue
+            caption = " ".join(self.text_content(c) for c in fig_captions.getchildren())
+            yield "Figure Caption", caption
+
+        # Table captions
+        tables = self.content.xpath("//body//table-wrap") or []
+        for table in tables:
+            caption_element = table.find("caption/p") or table.find("caption/title")
+            caption = self.text_content(caption_element)
+            yield "Table Caption", caption
+
+    @staticmethod
+    def text_content(element: etree._Element) -> str:
```

Comment:
I think one might need to change the type to `Optional[etree._Element]`

Reply:
Done in f8b21a0. Thanks for catching this!